specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json

```diff
@@ -880,6 +895,48 @@
         "hostNamePrefix": {
           "description": "Prefix for the hostName of the SignalR service. Retained for future use.\r\nThe hostname will be of format: &lt;hostNamePrefix&gt;.service.signalr.net.",
           "type": "string"
+        },
+        "features": {
+          "description": "List of SignalR featureFlags. e.g. ServiceMode.\r\n\r\nFeatureFlags that are not included in the parameters for the update operation will not be modified.\r\nAnd the response will only include featureFlags that are explicitly set. \r\nWhen a featureFlag is not explicitly set, SignalR service will use its globally default value. \r\nBut keep in mind, the default value doesn't mean \"false\". It varies in terms of different FeatureFlags.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SignalRFeature"
+          }
+        }
+      }
+    },
+    "SignalRFeature": {
+      "description": "Feature of a SignalR resource, which controls the SignalR runtime behavior.",
+      "required": [
+        "flag",
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "flag": {
+          "description": "Name of the feature. Required.",
+          "enum": [
+            "ServiceMode"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "FeatureFlags",
+            "modelAsString": true
+          }
+        },
+        "value": {
+          "description": "Value of the feature flag. See Azure SignalR service document https://docs.microsoft.com/en-us/azure/azure-signalr/ for allowed values.",
+          "maxLength": 128,
+          "minLength": 1,
+          "type": "string"
+        },
+        "properties": {
+          "description": "Optional properties related to this feature.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "x-ms-client-flatten": false
```

Comment:
nit: This doesn't need to be set explicitly

Reply:
Removed. This produces a warning:
```
WARNING (AvoidNestedProperties/R2001/SDKViolation): Consider using x-ms-client-flatten to provide a better end user experience
    - file:///mnt/d/workspace/github/juniwang/azure-rest-api-specs/specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json:924:8 ($.definitions.SignalRFeature.properties.properties)
```
If it doens't matter, we may suppress the warning in future.

Reply:
I see, we discourage too many levels of nesting especially on input parameters hence the warning. I don't think suppressing this warning is a good idea, please consider modelling the service in a way that it does not have nesting in properties like so. 
Not a blocker